stepman/git.go

```diff
@@ -24,18 +26,42 @@ func DoGitClone(uri, pth string) error {
 	return RunCommand("git", []string{"clone", "--recursive", uri, pth}...)
 }
 
-// DoGitCloneWithVersion ...
-func DoGitCloneWithVersion(uri, pth, version string) error {
+// DoGitCheckout ...
+func DoGitCheckout(commithash string) error {
+	if commithash == "" {
+		return errors.New("Git Clone 'hash' missing")
+	}
+	return RunCommand("git", []string{"checkout", commithash}...)
+}
+
+// DoGitCloneWithCommit ...
+func DoGitCloneWithCommit(uri, pth, version, commithash string) error {
 	if uri == "" {
 		return errors.New("Git Clone 'uri' missing")
 	}
 	if pth == "" {
 		return errors.New("Git Clone 'pth' missing")
 	}
-	if version == "" {
-		return errors.New("Git Clone 'version' missing")
+	if commithash == "" {
+		return errors.New("Git Clone 'hash' missing")
+	}
+	if err := RunCommand("git", []string{"clone", "--recursive", uri, pth, "--branch", version}...); err != nil {
+		return err
+	}
```

Comment:
cleanup (remove the git dir) if anything else after this fails - ex if the `hash` check fails. Do it with a `defer` block?